.github/workflows/ci-cd-prod.yml

```
@@ -0,0 +1,72 @@
+name: [production] CI/CD
+
+on:
+  push:
+    branches:
+      - main  # main 브랜치 기준으로 실행
+
+
+jobs:
+  build-docker-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    env:
+      profile: prod
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Build with Gradle Wrapper
+        run: ./gradlew clean build -x test
+
+      - name: Docker Image Build
+        run: docker build -t ${{ secrets.DOCKER_USERNAME }}/berich:latest .
```

latest 태그만 사용하면 이미지 변경을 추척하거나 롤백, 캐시 문제 발생이 일어날 수 있기에 이미지 태그 전략을 도입하는 것이 좋아보입니다.

```suggestion
- name: Set image tag
  run: echo "IMAGE_TAG=$(date +%Y%m%d-%H%M%S)-${GITHUB_SHA::7}" >> $GITHUB_ENV
  
...

docker build -t ${{ secrets.DOCKER_USERNAME }}/berich:${{ env.IMAGE_TAG }} .
```

---

오 제안해주신 방안 좋은 것 같습니다